src/pre_process/m_perturbation.fpp

```diff
@@ -606,4 +616,290 @@ contains
 
     end subroutine s_generate_wave
 
+    subroutine s_elliptic_smoothing(q_prim_vf)
+
+        type(scalar_field), dimension(sys_size), intent(INOUT) :: q_prim_vf
+        integer :: i, j, k, l, q
+
+        do q = 1, elliptic_smoothing_iters
+
+            if (bcxb >= 0) then
+                call s_mpi_sendrecv_variables_buffers(q_prim_vf, 1, -1)
+            else if (bcxb == -1) then
+                !$acc parallel loop gang vector collapse(3) default(present)
+                do l = 0, p
+                    do k = 0, n
+                        do j = 1, buff_size
+                            do i = 1, sys_size
+                                q_prim_vf(i)%sf(-j, k, l) = q_prim_vf(i)%sf(m - j + 1, k, l)
+                            end do
+                        end do
+                    end do
+                end do
+            else if (bcxb == -2) then
```

Comment:
this can't be the best way to do this...

Reply:
Addressed